server/CQuery.cpp

```diff
@@ -82,18 +83,46 @@ void CQuery::notifyObjectAboutRemoval(const CObjectVisitQuery &objectVisit) cons
 {
 }
 
-void CQuery::onExposure(CGameHandler *gh, QueryPtr topQuery)
+void CQuery::onExposure(QueryPtr topQuery)
 {
-	gh->queries.popQuery(*this);
+	logGlobal->trace("Exposed query with id %d", queryID);
+	owner->popQuery(*this);
 }
 
-void CQuery::onAdding(CGameHandler *gh, PlayerColor color)
+void CQuery::onAdding(PlayerColor color)
 {
 
 }
 
-CObjectVisitQuery::CObjectVisitQuery(const CGObjectInstance *Obj, const CGHeroInstance *Hero, int3 Tile)
-	: visitedObject(Obj), visitingHero(Hero), tile(Tile), removeObjectAfterVisit(false)
+void CQuery::onAdded(PlayerColor color)
+{
+
+}

```

Comment:
Might be we should decide whatever we add one extra line into empty methods or no.